File: Exercise/LuoGu/Code_2023_7_18/P1047.cpp
#include <iostream>
#include <algorithm>
#include <math.h>
#include <vector>
#include <string>
#include <string.h>

using namespace std;

int main(void)
{
    int l, m;
    cin >> l >> m;
    int line[l + 1];

    memset(line, 0, l + 1);

    int start, end;

    while (m--)
    {
        cin >> start >> end;
        for (int i = start; i <= end; i++)
        {
            line[i] = -1;
        }
    }

    int cnt = 0;
    for (int i = 0; i <= l; i++)
    {
        if (line[i] != -1)
        {
            cnt++;
        }
    }

    cout << cnt << endl;

    return 0;
}

File: Exercise/LuoGu/Code_2023_7_18/P1047.java
package Code_2023_7_18;

import java.util.Scanner;

public class P1047 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int l = sc.nextInt();
        int[] line = new int[l + 1];
        int m = sc.nextInt();
        int start, end;
        while (m-- > 0) {
            start = sc.nextInt();
            end = sc.nextInt();
            for (int i = start; i <= end; i++) {
                line[i] = -1;
            }
        }

        int cnt = 0;
        for (int i = 0; i <= l; i++) {
            if (line[i] != -1) {
                cnt++;
            }
        }

        System.out.println(cnt);
        sc.close();
    }
}
